tests/backup/backup_locked_bucket_test.go

```diff
@@ -412,3 +412,241 @@ var _ = Describe("{LockedBucketResizeOnRestoredVolume}", func() {
+		contexts                    []*scheduler.Context
+		preRuleNameList             []string
+		postRuleNameList            []string
+		appContexts                 []*scheduler.Context
+		appNamespaces               []string
+		clusterStatus               api.ClusterInfo_StatusInfo_Status
+		beforeSize                  int
+		credName                    string
+		periodicSchedulePolicyName  string
+		periodicSchedulePolicyNames []string
+		periodicSchedulePolicyUid   string
+		periodicSchedulePolicyUids  []string
+		scheduleName                string
+		cloudCredUID                string
+		backupLocation              string
+	)
+	labelSelectors := make(map[string]string)
+	cloudCredUIDMap := make(map[string]string)
+	backupLocationMap := make(map[string]string)
+	podListBeforeSizeMap := make(map[string]int)
+	podListAfterSizeMap := make(map[string]int)
+	contexts = make([]*scheduler.Context, 0)
+	appNamespaces = make([]string, 0)
+	modes := [2]string{"GOVERNANCE", "COMPLIANCE"}
+	JustBeforeEach(func() {
+		StartTorpedoTest("LockedBucketResizeVolumeOnScheduleBackup", "Verify schedule backups are successful while volume resize is in progress for locked bucket", nil, 59899)
+		log.InfoD("Verifying if the pre/post rules for the required apps are present in the list or not")
+		for i := 0; i < len(appList); i++ {
+			if Contains(postRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["post_action_list"]; ok {
+					dash.VerifyFatal(ok, true, fmt.Sprintf("Post Rule details mentioned for the app %v", appList[i]))
+				}
+			}
+			if Contains(preRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["pre_action_list"]; ok {
+					dash.VerifyFatal(ok, true, fmt.Sprintf("Pre Rule details mentioned for the app %v", appList[i]))
+				}
+			}
+		}
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+			}
+		}
+	})
+	It("Schedule backup while resizing the volume", func() {
+		providers := getProviders()
+		Step("Validate applications", func() {
+			ValidateApplications(contexts)
+		})
+		Step("Creating rules for backup", func() {
+			log.InfoD("Creating pre and post rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating pre rule for deployed apps for %v with status %v", appList[i], preRuleStatus))
+				dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+				preRuleNameList = append(preRuleNameList, ruleName)
```

Comment:
How are we calling these rule's here in createbackup if we have mutiple apps ?

Reply:
Done